File: packages/contracts/src/types/report.ts
import type {
  AuditReceipt,
  DetectedRegion,
  Finding,
  ImageMetadata,
  ImageQualityReport,
  ModelInfo
} from "./analysis";

export type ExportFormat = "json" | "html" | "evidence_map" | "audit_receipt";

export interface ReportSection {
  readonly sectionId: string;
  readonly title: string;
  readonly summary: string;
  readonly findingIds: readonly string[];
}

export interface EvidenceMapRegion {
  readonly regionId: string;
  readonly label: string;
  readonly bbox: DetectedRegion["bbox"];
  readonly confidence: number;
  readonly evidenceRefs: readonly string[];
}

export interface EvidenceMap {
  readonly schemaVersion: "1.0.0";
  readonly runId: string;
  readonly imageId: string;
  readonly width: number;
  readonly height: number;
  readonly regions: readonly EvidenceMapRegion[];
  readonly generatedAt: string;
}

export interface VisualReport {
  readonly schemaVersion: "1.0.0";
  readonly reportId: string;
  readonly runId: string;
  readonly title: string;
  readonly generatedAt: string;
  readonly image: ImageMetadata;
  readonly qualityReport: ImageQualityReport;
  readonly models: readonly ModelInfo[];
  readonly sections: readonly ReportSection[];
  readonly findings: readonly Finding[];
  readonly evidenceMap: EvidenceMap;
  readonly auditReceipt: AuditReceipt;
}

File: engine/veriframe_core/veriframe_core/contracts/report.py
from __future__ import annotations

from typing import Literal

from pydantic import Field

from veriframe_core.contracts.analysis import (
    AuditReceipt,
    BoundingBox,
    Finding,
    ImageMetadata,
    ImageQualityReport,
    ModelInfo,
    StrictModel,
)

ExportFormat = Literal["json", "html", "evidence_map", "audit_receipt"]


class ReportSection(StrictModel):
    sectionId: str = Field(min_length=1)
    title: str = Field(min_length=1)
    summary: str
    findingIds: list[str]


class EvidenceMapRegion(StrictModel):
    regionId: str = Field(min_length=1)
    label: str = Field(min_length=1)
    bbox: BoundingBox
    confidence: float = Field(ge=0, le=1)
    evidenceRefs: list[str]


class EvidenceMap(StrictModel):
    schemaVersion: Literal["1.0.0"]
    runId: str = Field(min_length=1)
    imageId: str = Field(min_length=1)
    width: int = Field(ge=1)
    height: int = Field(ge=1)
    regions: list[EvidenceMapRegion]
    generatedAt: str


class VisualReport(StrictModel):
    schemaVersion: Literal["1.0.0"]
    reportId: str = Field(min_length=1)
    runId: str = Field(min_length=1)
    title: str = Field(min_length=1)
    generatedAt: str
    image: ImageMetadata
    qualityReport: ImageQualityReport
    models: list[ModelInfo]
    sections: list[ReportSection]
    findings: list[Finding]
    evidenceMap: EvidenceMap
    auditReceipt: AuditReceipt
